vue/sbc-common-components/src/components/SbcHeader.vue

```diff
@@ -454,7 +455,14 @@ export default class SbcHeader extends Mixins(NavigationMixin) {
   }
 
   private goToCreateBCSCAccount () {
-    this.redirectToPath(this.inAuth, Pages.CREATE_ACCOUNT)
+    if (this.inAuth) {
+      this.$router.push(`/${Pages.CREATE_ACCOUNT}`)
+    } else {
+      // To send the navigateBackUrl to duplicate account creator interceptor screen, so that we can redirect to that page.
+      let redirectUrl = new URL(`${ConfigHelper.getAuthContextPath()}/${Pages.CREATE_ACCOUNT}`)
+      if (this.dashboardReturnUrl) redirectUrl.searchParams.append('redirectToUrl', encodeURIComponent(this.dashboardReturnUrl))
+      window.location.assign(redirectUrl.toString())
+    }
```

Comment:
Can you please explain what impacts this has to the UIs that use this library. What flow(s) does this impact, and how? Do the UIs need to update anything, or provide new props?

Reply:
Hi. As part of 6921. the idea is to intercept user with a duplicate account warning screen when the user already has an existing account and try to create another account. 
Therefore, when the user clicks create account, 

- In the sbc-common:(sbc-header) component, we capture the redirect url link ( as a property ) and direct to setup-account of sbc-auth. 
- In the sbc-auth, when the user decides not to create account and tries to access existing account, we route to the redirect url that we captured earlier.
- A default url is also used incase there is no redirect url. ( sbc-auth home)
- For you reference, we have created a dependency ticket as well https://github.com/bcgov/entity/issues/7757
- And a PR in sbc-auth as well for your reference https://github.com/bcgov/sbc-auth/pull/1646

Reply:
So the other UIs (Filings UIs, Create UI, Edit UI, Namerequest UI and PPR UI) don't need to change anything (except to import the latest sbc-common-components) ?

Reply:
And to pass the redirect url in the **dashboardReturnUrl** property in the Sbc-Header value so that we can direct the user to that link when the user selects access account button.
![image](https://user-images.githubusercontent.com/78374865/120229950-249ea800-c203-11eb-94df-9e809e4d8463.png)

Reply:
Are you saying each UI needs to pass in this new prop?

Could this prop be defaulted from the current session storage variables (so that each UI does not have to populate the common variables)?

Reply:
we can pass the dashboardReturnUrl as a property as follows

```
<sbc-header
        :key="$store.state.refreshKey"
        :in-auth="true"
        :show-product-selector="false"
        :dashboardReturnUrl="dashboardReturnUrl"
        @account-switch-started="startAccountSwitch"
        @account-switch-completed="completeAccountSwitch"
        @hook:mounted="setup"
        ref="header" :redirect-on-logout="logoutUrl">
        <template v-slot:login-button-text>
          Log in with BC Services Card
        </template>
      </sbc-header>
```

Reply:
OK, thanks.

-> Do the UIs need to pass this new prop?

Reply:
We are defaulting it to Sbc-auth /home route incase there is no prop passed, but we can pass the value as well :)

Reply:
I still don't understand if the UIs (Filings, Namerequest, etc) are **required** to pass in the new prop or not.

Reply:
Hi @severinbeauvais the dependency ticket https://github.com/bcgov/entity/issues/7757 will come to your way soon... This is not a mandatory param , If you need any behaviour other than the default behaviour , the url could be passed in as a prop. That decision will be a discussion with in each team .
So its not required , But if you want customization , you can pass the property.
This is not an every day behaviour which user will going through , rather this is a preventive measure incase if user accidentally or unintentionally clicks on create account OR somehow has a hard time finding the current account ..

Reply:
Thanks Saravan. That's exactly what I wanted to know 👍